stories/select.story.tsx

```diff
@@ -74,56 +82,47 @@ storiesOf("Components/Select", module)
       )
     )
   )
-  .add("Multi Select", () => (
-    <Select
-      onChange={action("onSelect")}
-      placeholder="Choose Option"
-      multiSelect
-      onApply={action("onApply")}
-      onClear={action("onClear")}
-      selected={undefined} // TODO:Aziz add withState
-    >
-      {new Array(20).fill(1).map((_x, i) => (
-        <Option
-          key={i + 1}
-          value={`option-${i + 1}`}
-          label={`I am an option-${i + 1}`}
-        />
-      ))}
-    </Select>
-  ))
   .add(
-    "Multi Select with searchbox",
-    withState<State>({ searchQuery: "", multiSelected: undefined })(
+    "Multi Select",
+    withState<MultiSelectState>({ searchQuery: "", selected: [], value: "" })(
       ({ store }) => (
-        <Select
-          onChange={val => {
+        <Select<OptionType>
+          multiSelect
+          selected={store.state.selected}
+          onChange={(val: Array<unknown>) => {
             action("onSelect")(val);
-            store.set({ multiSelected: val as number[] | string[] });
+            store.set({ selected: val as OptionType[] });
           }}
           placeholder="Choose Option"
-          multiSelect
-          searchBox
-          selected={store.state.multiSelected} // TODO:Aziz add withState
+          searchBox={boolean("Search Box", true)}
           onApply={action("onApply")}
           onClear={action("onClear")}
           searchBoxProps={{
             value: store.state.searchQuery,
-            onChange: query => {
+            onChange: (query: string) => {
               action("searchBoxProps.onChange")(query);
               store.set({ searchQuery: query });
             },
             placeholder: "Search",
             clearable: boolean("searchBoxProps.clearable", true)
           }}
         >
-          {new Array(20).fill(1).map((_x, i) => (
-            <Option
-              key={i + 1}
-              value={`option-${i + 1}`}
-              label={`I am an option-${i + 1}`}
-            />
-          ))}
+          {options
+            .filter(option => {
+              if (!boolean("Search Box", true) || !store.state.searchQuery) {
+                return true;
+              }
+              return option.label
+                .toLowerCase()
+                .includes(store.state.searchQuery.toLowerCase());
+            })
+            .map(option => (
+              <Option<OptionType>
+                key={option.id}
+                value={option}
```

Comment:
I guess this is what you wanted @dxtr026

Reply:
Yes